data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java

```diff
@@ -234,15 +235,18 @@ private void addExceptionHandlers() {
     app.exception(BadRequestException.class, this::badRequest);
     app.exception(JsonProcessingException.class, this::badRequest);
     app.exception(IllegalArgumentException.class, this::badRequest);
-    // Add catch-all handler
-    app.exception(
-        Exception.class,
-        (e, ctx) -> {
-          LOG.error("Failed to process request to URL {}", ctx.url(), e);
-          ctx.status(SC_INTERNAL_SERVER_ERROR);
-          setErrorBody(
-              ctx, () -> BadRequest.internalError(jsonProvider, "An unexpected error occurred"));
-        });
+    app.exception(Exception.class, this::catchAllExceptionHandler);
+  }
+
+  private void catchAllExceptionHandler(final Throwable throwable, final Context context) {
+    if (throwable.getCause() instanceof EOFException) {
```

Comment:
I think we need to check if anything in the cause chain is `EOFException` as it may wind up being wrapped multiple times.  So use `ThrowableUtil.getCause(throwable, EOFException.class).isPresent()` (or maybe add a `ThrowableUtil.hasCause(throwable, EOFException.class)` that just returns boolean for simplicity.